ironfish/src/evm/evm.ts

```diff
@@ -39,4 +42,34 @@ export class IronfishEvm {
   async runTx(opts: RunTxOpts): Promise<RunTxResult> {
     return this.vm.runTx(opts)
   }
+
+  async verifyTx(opts: RunTxOpts): Promise<EvmResult> {
+    // TODO(jwp) add db transaction and roll back
+    const result = await this.runTx(opts)
+
+    // TODO(jwp) from custom opcodes populate shields and unshields
```

Comment:
Not sure how we will capture the opcode events during this call, but this is where it will go